test/common/router/scoped_config_impl_test.cc

```diff
@@ -49,8 +49,10 @@ TEST(StringKeyFragmentTest, Normal) {
 
 TEST(HeaderValueExtractorImplDeathTest, InvalidConfig) {
   ScopedRoutes::ScopeKeyBuilder::FragmentBuilder config;
-  // Type not set.
-  EXPECT_DEATH(HeaderValueExtractorImpl{config}, "header_value_extractor is not set.");
+  // Type not set, ASSERT only fails in debug mode.
+#if !defined(NDEBUG)
```

Comment:
The issue here is that the constructor fails differently when `NDEBUG` is defined (it throws a `ProtoValidationException`) and when it's not (it aborts due to the assertion failure).

I would suggest either checking both and adding a comment for clarity, or avoiding the ambiguity and failing the same way in both opt and debug.

Reply:
done